### PR TITLE
feat: warn user about incoming expiration of session

### PR DESCRIPTION
**Pull request by beawar**
Show some snackbar to let know user the session is expiring soon. Once session is expired, redirect user to login page by performing a programmatic logout.

Refs: SHELL-207

### CODE REVIEW

**Review by gnekoz — DISMISSED** [review]

File: src/boot/loader.tsx [hunk 1]
```diff
@@ -105,6 +117,78 @@ export const Loader = (): React.JSX.Element => {
...
+						label: t(
+							'snackbar.expiration.threeMinutes',
+							"Your session will expire in 60 seconds. After that, you'll be redirected to the login page."
+						),
+						actionLabel: t('snackbar.expiration.action', 'Go to login page'),
+						onActionClick: logoutFn,
+						replace: true
+					});
+					expirationTimeouts.push(setTimeout(logoutFn, Math.min(oneMinute, sessionLifetime)));
+				}, sessionLifetime - oneMinute)
```
Comment (CataldoMazzilli) [inline src/boot/loader.tsx]: just to be safe...can we wrap in a Math.max(0, sessionLifetime - oneMinute)?

File: src/boot/loader.tsx [hunk 2]
```diff
@@ -105,6 +117,78 @@ export const Loader = (): React.JSX.Element => {
 		return () => {
 			unloadAllApps();
 		};
-	}, []);
+	}, [getSessionInfo]);
+
+	useEffect(() => {
+		const expirationTimeouts: NodeJS.Timeout[] = [];
+		const logoutFn = (): void => {
```
Comment (CataldoMazzilli) [inline src/boot/loader.tsx:124]: can we use directly logout?

Comment (beawar) [reply src/boot/loader.tsx:124]: Using logout directly cause the sonarlint error typescript:S6544
SonarLint: Promise-returning function provided to property where a void return was expected.

That's why I wrapped it in a non-promise-returning function

**Review by CataldoMazzilli — DISMISSED** [review]

**Review by CataldoMazzilli — DISMISSED** [review]

File: src/boot/loader.tsx [hunk 3]
```diff
@@ -105,6 +117,81 @@ export const Loader = (): React.JSX.Element => {
...
+			const oneMinute = 60 * 1000;
+			expirationTimeouts.push(
+				setTimeout(
+					() => {
+						createSnackbar({
+							severity: 'warning',
+							key: 'one-minute-from-expiration-snackbar',
+							autoHideTimeout: Math.min(oneMinute, sessionLifetime),
+							label: t(
+								'snackbar.expiration.threeMinutes',
```
Comment (beawar) [inline src/boot/loader.tsx]:
```suggestion
								'snackbar.expiration.oneMinute',
```

**Review by CataldoMazzilli — APPROVED** [review]

**Review by rodleyorosa — APPROVED** [review]

**Review by CerDennis — APPROVED** [review]

**Pull request merged**